cmake/FindELPA.cmake

```diff
@@ -7,34 +7,46 @@
 #  ELPA_INCLUDE_DIR - Where to find ELPA headers.
 #
 
-find_path(ELPA_INCLUDE_DIR
+find_package(PkgConfig)
+
+if(PKG_CONFIG_FOUND)
+        if(USE_OPENMP)
+    pkg_search_module(ELPA REQUIRED IMPORTED_TARGET GLOBAL elpa_openmp)
+  else()
+    pkg_search_module(ELPA REQUIRED IMPORTED_TARGET GLOBAL elpa)
+  endif()
+else()
+  find_path(ELPA_INCLUDE_DIRS
     elpa/elpa.h
     HINTS ${ELPA_DIR}
     PATH_SUFFIXES "include" "include/elpa"
     )
-if(USE_OPENMP)
-    find_library(ELPA_LIBRARY
-        NAMES elpa_openmp elpa
-        HINTS ${ELPA_DIR}
-        PATH_SUFFIXES "lib"
-        )
-else()
-    find_library(ELPA_LIBRARY
-        NAMES elpa
-        HINTS ${ELPA_DIR}
-        PATH_SUFFIXES "lib"
-        )
+  if(USE_OPENMP)
+    find_library(ELPA_LINK_LIBRARIES
+      NAMES elpa_openmp elpa
+      HINTS ${ELPA_DIR}
+      PATH_SUFFIXES "lib"
+      )
+  else()
+    find_library(ELPA_LINK_LIBRARIES
+      NAMES elpa
+      HINTS ${ELPA_DIR}
+      PATH_SUFFIXES "lib"
+      )
+  endif()
+  #message(
+  #  "ELPA : We need pkg-config to get all information about the elpa library")
 endif()
 
 # Handle the QUIET and REQUIRED arguments and
 # set ELPA_FOUND to TRUE if all variables are non-zero.
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(ELPA DEFAULT_MSG ELPA_LIBRARY ELPA_INCLUDE_DIR)
+find_package_handle_standard_args(ELPA DEFAULT_MSG ELPA_LINK_LIBRARIES ELPA_INCLUDE_DIRS)
 
 # Copy the results to the output variables and target.
 if(ELPA_FOUND)
-    set(ELPA_LIBRARIES ${ELPA_LIBRARY})
-    set(ELPA_INCLUDE_DIR ${ELPA_INCLUDE_DIR})
+    set(ELPA_LIBRARY ${ELPA_LINK_LIBRARIES})
+    set(ELPA_INCLUDE_DIR ${ELPA_INCLUDE_DIRS})
```

Comment:
Does changing the name affect the linking process in main CMakeLists.txt file?

Reply:
The names used in CMakeFiles/outer FindELPA.cmake is not affected. Here ELPA_LIBRARY and ELPA_INCLUDE_DIR are the previous names, and ELPA_LINK_LIBRARIES/ELPA_INCLUDE_DIRS are default return variable names of pkg_search_module; here value of new vars are assigned to old vars, so that they still can be used outer FindELPA.cmake.